doc/contribute/committer_expectations.rst

```diff
@@ -0,0 +1,272 @@
+.. _committer-expectations:
+
+Committer Expectations
+######################
+
+Overview
+********
+
+The Zephyr project encourages you to submit changes as small pull requests.
+Small pull requests (PRs) have the following benefits:
+
+- Reviewed more quickly and reviewed more thoroughly. It's easier for reviewers
+  to set aside a few minutes to review small changes several times than it is to
+  allocate large block of time review a large PR.
+
+- Less wasted work if reviewers or maintainers reject the direction of the
+  change.
+
+- Easier to rebase and merge. Smaller PRs are less likely to conflict with other
+  changes in the tree.
+
+- Easier to revert if the PR breaks functionality.
+
+
+Defining Small PRs
+==================
+
+- Small PRs should encompass one self-contained logical change.
+
+- When adding a new large feature or API, the PR should address only one part of
+  the feature. In this case create an :ref:`RFC proposal <rfcs>` [#rfc]_ to
+  describe the additional parts of the feature for reviewers.
+
+- PRs should include tests or samples for any added or changed lines of code.
+
+- PRs must update any documentation affected by the functional code changes.
+
+- If introducing a new API, the PR must include an example usage of the API.
+  This provides context to the reviewer and prevents submitting PRs with unused
+  APIs.
+
+
+Multiple Commits on a Single PR
+===============================
+
+You are further encouraged to break up PRs into multiple commits.  Keep in mind
+each commit of your PR must still build cleanly and pass all the CI tests.
+
+For example, when introducing an extension to an API, you can break up the PR
+into multiple commits targeting these specific changes:
+
+#. Introduce the new APIs, including shared devicetree bindings
+#. Update driver implementation X, with driver specific devicetree bindings
+#. Update driver implementation Y
+#. Add tests for the new API
+#. Add a sample using the API
+#. Update the documentation
+
+Large Changes
+=============
+
+Large changes to the Zephyr project must submit an :ref:`RFC proposal <rfcs>`
+describing the full scope of change and future work.  The RFC proposal provides
+the required context to reviewers, but allows for smaller, incremental, PRs to
+get reviewed and merged into the project. The RFC should also define the minimum
+viable implementation.
```

Comment:
"It is possible to share and discuss early code supporting RFCs in large, temporary draft PRs that are  too large to be merged as is but intended to be broken down in smaller PRs later."